src/scripts/download-new-songs.ts

```diff
@@ -187,23 +184,49 @@ const downloadNewSongs = async (limit?: number) => {
             deadLinksSkipped++;
             continue;
         }
-        try {
-            logger.info(`Downloading song: '${song.name}' by ${song.artist} | ${song.youtubeLink} (${downloadCount + 1}/${songsToDownload.length})`);
-            const mp3Path = await downloadSong(song.youtubeLink);
-            logger.info(`Encoding song: '${song.name}' by ${song.artist} | ${song.youtubeLink}`);
+
+        const retryJob = async (job: (songID: string) => Promise<void>): Promise<void> => new Promise(async (resolve, reject) => {
```

Comment:
Move this to `utils`

Reply:
It should take in any arbitrary function, not specifically one that consumes a songID

Reply:
Make it a regular function instead of an arrow function